### PR TITLE
Fix rasterizing a Path with categorical values including empty strings

### PR DESCRIPTION
**Pull request by maximlt**
Fixes https://github.com/holoviz/holoviews/issues/6326

### CODE REVIEW

File: holoviews/operation/datashader.py [hunk 1]
```diff
@@ -448,6 +448,12 @@ def _apply_datashader(self, dfdata, cvs_fn, agg_fn, agg_kwargs, x, y):
                     val = val.astype(np.float64)
                     val[neg1] = np.nan
                 agg[col] = ((y.name, x.name), val)
+
+        if isinstance(agg_fn, ds.count_cat):
```
Comment (maximlt) [inline holoviews/operation/datashader.py:452]: Hopefully a good place to add this.

File: holoviews/tests/operation/test_datashader.py [hunk 2]
```diff
@@ -1296,6 +1296,42 @@ def test_rasterize_apply_when_instance_with_line_width(self):
...
+        #         [0, 0],
+        #         [0, 0],
+        #         [0, 1]],
+        #     [[1, 0],
+        #         [0, 0],
+        #         [0, 0],
+        #         [0, 1]],
+        # ])
+        # expected = ImageStack((x, y, array), vdims=['a', 'b'])
+        # self.assertEqual(rasterized, expected)
```
Comment (maximlt) [inline holoviews/tests/operation/test_datashader.py:1333]: @hoxbro I commented out this code for now as I realized that `compare.py` has not been updated to compare `ImageStack` objects part of the test suite. Is this intentional? Not that this part of the test is not strictly required, I worked on this fix together with @azaya89 and ended up showing him that `self.assertEqual` is extended by HoloViews. Ultimately, the real check is that there's no error raised and no empty string in the vdims.

Comment (hoxbro) [reply holoviews/tests/operation/test_datashader.py:1333]: It was not intentional. At some point, I want to transition away from using the stdlib unittest framework.  

Can you uncomment this and do some simple `assert` instead?

File: holoviews/operation/datashader.py (hunk 1 above)
Comment (hoxbro) [reply holoviews/operation/datashader.py:452]: Placement looks fine to me. We also need this for `ds.by`:
```suggestion
        if isinstance(agg_fn, (ds.count_cat, ds.by)):
```

We could parameterize the test.

**Pull request merged**